### ISSUE TITLE
train normal but val map all 0

### ISSUE THREAD

**Issue by huihui6666** [AILab-CVC/YOLO-World#531]
Hi,I finetune in my dataset,the loss of train drop from 150 to 59,but the val map is all 0

**Comment by huihui6666** [comment]
I use the config file `yolo_world_v2_l_efficient_neck_2e-4_80e_8gpus_mask-refine_finetune_coco.py`

**Comment by huihui6666** [comment]
some of my train log as follow
2024/11/26 11:33:52 - mmengine - INFO - Exp name: yolo_world_v2_m_vlpan_bn_2e-4_80e_8gpus_mask-refine_finetune_coco_my_20241126_091045
2024/11/26 11:34:01 - mmengine - INFO - Epoch(train) [35][7050/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:35:59  time: 0.2570  data_time: 0.0024  memory: 7823  grad_norm: 127.9921  loss: 59.7234  loss_cls: 22.0544  loss_bbox: 17.9890  loss_dfl: 19.6800
2024/11/26 11:34:14 - mmengine - INFO - Epoch(train) [35][7100/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:01  time: 0.2550  data_time: 0.0025  memory: 7570  grad_norm: 122.7781  loss: 58.5399  loss_cls: 21.2850  loss_bbox: 17.7463  loss_dfl: 19.5087
2024/11/26 11:34:27 - mmengine - INFO - Epoch(train) [35][7150/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:06  time: 0.2585  data_time: 0.0025  memory: 7476  grad_norm: 136.3488  loss: 59.4846  loss_cls: 22.1040  loss_bbox: 17.7713  loss_dfl: 19.6094
2024/11/26 11:34:40 - mmengine - INFO - Epoch(train) [35][7200/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:10  time: 0.2595  data_time: 0.0024  memory: 7890  grad_norm: 123.3398  loss: 58.9759  loss_cls: 21.3812  loss_bbox: 18.1015  loss_dfl: 19.4932
2024/11/26 11:34:53 - mmengine - INFO - Epoch(train) [35][7250/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:17  time: 0.2636  data_time: 0.0025  memory: 7543  grad_norm: 126.9679  loss: 58.8478  loss_cls: 21.6973  loss_bbox: 17.7954  loss_dfl: 19.3551
2024/11/26 11:35:06 - mmengine - INFO - Epoch(train) [35][7300/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:18  time: 0.2523  data_time: 0.0024  memory: 7516  grad_norm: 142.3983  loss: 59.2025  loss_cls: 21.8677  loss_bbox: 17.7598  loss_dfl: 19.5749
2024/11/26 11:35:18 - mmengine - INFO - Epoch(train) [35][7350/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:13  time: 0.2399  data_time: 0.0025  memory: 7743  grad_norm: 134.2802  loss: 59.7969  loss_cls: 22.3881  loss_bbox: 17.8545  loss_dfl: 19.5544
2024/11/26 11:35:29 - mmengine - INFO - Epoch(train) [35][7400/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:36:01  time: 0.2236  data_time: 0.0024  memory: 7504  grad_norm: 131.0194  loss: 58.4260  loss_cls: 21.4178  loss_bbox: 17.6953  loss_dfl: 19.3128
2024/11/26 11:35:40 - mmengine - INFO - Epoch(train) [35][7450/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:35:44  time: 0.2118  data_time: 0.0024  memory: 7690  grad_norm: 126.3277  loss: 59.0065  loss_cls: 21.8161  loss_bbox: 17.7280  loss_dfl: 19.4624
2024/11/26 11:35:51 - mmengine - INFO - Epoch(train) [35][7500/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:35:30  time: 0.2173  data_time: 0.0023  memory: 7410  grad_norm: 127.3711  loss: 59.7367  loss_cls: 22.2691  loss_bbox: 17.9271  loss_dfl: 19.5405
2024/11/26 11:36:02 - mmengine - INFO - Epoch(train) [35][7550/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:35:16  time: 0.2196  data_time: 0.0024  memory: 7543  grad_norm: 143.3271  loss: 59.3853  loss_cls: 22.1371  loss_bbox: 17.8032  loss_dfl: 19.4451
2024/11/26 11:36:13 - mmengine - INFO - Epoch(train) [35][7600/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:35:03  time: 0.2201  data_time: 0.0024  memory: 7664  grad_norm: 126.6539  loss: 58.3427  loss_cls: 21.2495  loss_bbox: 17.7076  loss_dfl: 19.3856
2024/11/26 11:36:24 - mmengine - INFO - Epoch(train) [35][7650/7676]  base_lr: 2.0000e-04 lr: 1.1833e-04  eta: 21:34:48  time: 0.2168  data_time: 0.0024  memory: 7436  grad_norm: 132.4634  loss: 58.1278  loss_cls: 21.3929  loss_bbox: 17.3028  loss_dfl: 19.4322
2024/11/26 11:36:29 - mmengine - INFO - Exp name: yolo_world_v2_m_vlpan_bn_2e-4_80e_8gpus_mask-refine_finetune_coco_my_20241126_091045
2024/11/26 11:36:29 - mmengine - INFO - Saving checkpoint at 35 epochs

2024/11/26 11:40:20 - mmengine - INFO - Epoch(val) [35][6100/6429]    eta: 0:00:12  time: 0.0351  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:21 - mmengine - INFO - Epoch(val) [35][6150/6429]    eta: 0:00:10  time: 0.0352  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:23 - mmengine - INFO - Epoch(val) [35][6200/6429]    eta: 0:00:08  time: 0.0338  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:25 - mmengine - INFO - Epoch(val) [35][6250/6429]    eta: 0:00:06  time: 0.0365  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:27 - mmengine - INFO - Epoch(val) [35][6300/6429]    eta: 0:00:04  time: 0.0343  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:28 - mmengine - INFO - Epoch(val) [35][6350/6429]    eta: 0:00:02  time: 0.0321  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:30 - mmengine - INFO - Epoch(val) [35][6400/6429]    eta: 0:00:01  time: 0.0336  data_time: 0.0003  memory: 1123  
2024/11/26 11:40:42 - mmengine - INFO - Evaluating bbox...
2024/11/26 11:41:52 - mmengine - INFO - bbox_mAP_copypaste: 0.000 0.001 0.000 0.000 0.001 0.000
2024/11/26 11:41:53 - mmengine - INFO - Epoch(val) [35][6429/6429]    coco/bbox_mAP: 0.0000  coco/bbox_mAP_50: 0.0010  coco/bbox_mAP_75: 0.0000  coco/bbox_mAP_s: 0.0000  coco/bbox_mAP_m: 0.0010  coco/bbox_mAP_l: 0.0000  data_time: 0.0004  time: 0.0367

**Comment by Vireakdara** [comment]
Did you found the solution yet ?

**Comment by yunfei-ai** [comment]
hi，我遇见了一个问题就是自定义数据集，将原有的coco类别80改为10类，结果出现了indexerror，我尝试了改变原有的类别数还是解决不了问题，最后我认为 self.cat_ids的维度小于你设置的类的数量导致的，可我却没有什么思路解决
